File: LeetCode/Easy/7/7.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == 0:
            return 0;
        elif x > 0:
            t = int(str(x)[::-1])
            if t > 2147483647:
                return 0
            else:
                return t
        else:
            t = int((str(x)[1:])[::-1])
            if t > 2147483647:
                return 0
            else:
                return t * -1

File: LeetCode/Easy/7/7.java
class Solution {
    public int reverse(int x) {
        StringBuffer r = null;
        if (x == 0)
            return 0;
        else if (x > 0) {
            r = new StringBuffer(Integer.toString(x));
            try {
                return Integer.parseInt((r.reverse().toString()));   
            } catch(NumberFormatException e) {
                return 0;
            }
        }
        else {
            r = new StringBuffer((Integer.toString(x)).substring(1));
            try {
                return (Integer.parseInt((r.reverse().toString())) * -1);
            } catch(NumberFormatException e) {
                return 0;
            }
        }
    }
}
